xarray/core/formatting.py

```diff
@@ -500,6 +500,13 @@ def diff_dim_summary(a, b):
 
 
 def _diff_mapping_repr(a_mapping, b_mapping, compat, title, summarizer, col_width=None):
+    def is_array_like(value):
+        return (
+            hasattr(value, "ndim")
+            and hasattr(value, "shape")
+            and hasattr(value, "dtype")
+        )
```

Comment:
Looks reasonable at the moment; unless anyone knows better?
This could move into utils if we get more confident (almost surprised we don't do this test elsewhere...)

Reply:
we have mostly been using `__array_function__` instead, but that is not really sufficient: `pint` defines `__array_function__` even if it wraps python scalars. In that case `pint.Quantity` is not array-like.